### PR TITLE
Past Pushes: fix error if alert field is not exist in payload

### PR DESCRIPTION
**Pull request by hermanliang**

### CODE REVIEW

File: src/dashboard/Push/PushIndex.react.js [hunk 1]
```diff
@@ -141,15 +141,11 @@ let getPushName = (pushData) => {
     );
   } else {
     let payload = pushData[PushConstants.PAYLOAD_FIELD];
```
Comment (peterdotjs) [inline src/dashboard/Push/PushIndex.react.js]: You could just change this line to:  `let payload = pushData[PushConstants.PAYLOAD_FIELD] || '';`

**Comment by facebook-github-bot-4** [comment]
@hermanliang updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@hermanliang updated the pull request.

File: src/dashboard/Push/PushIndex.react.js [hunk 2]
```diff
@@ -140,15 +140,15 @@ let getPushName = (pushData) => {
...
+    let payload = pushData[PushConstants.PAYLOAD_FIELD] || '';
     try {
       payload = JSON.parse(payload);
     } catch(e) { }
     if (payload) {
       if (typeof payload.alert === 'string') {
         return payload.alert;
       }
-      return payload.alert ? JSON.stringify(payload.alert) : payload;
+      return payload.alert ? JSON.stringify(payload.alert) : JSON.stringify(payload);
```
Comment (peterdotjs) [inline src/dashboard/Push/PushIndex.react.js]: Cool! One nit is that when the payload is empty string `JSON.stringify('')` results in a double string outputted. We'll only need stringify if payload is an object. Once you've added that check its go to merge. Thanks!

**Comment by facebook-github-bot-4** [comment]
@hermanliang updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@hermanliang updated the pull request.

File: src/dashboard/Push/PushIndex.react.js [hunk 3]
```diff
@@ -140,15 +140,17 @@ let getPushName = (pushData) => {
...
     } catch(e) { }
-    if (payload) {
+    if (typeof payload === 'object') {
       if (typeof payload.alert === 'string') {
         return payload.alert;
       }
-      return payload.alert ? JSON.stringify(payload.alert) : payload;
+      return payload.alert ? JSON.stringify(payload.alert) : JSON.stringify(payload);
+    } else {
+      return '';
```
Comment (hermanliang) [inline src/dashboard/Push/PushIndex.react.js]: If payload is not an object
`return ''`

File: src/dashboard/Push/PushIndex.react.js [hunk 4]
```diff
@@ -140,15 +140,17 @@ let getPushName = (pushData) => {
       <strong>{title}</strong>
     );
   } else {
-    let payload = pushData[PushConstants.PAYLOAD_FIELD];
+    let payload = pushData[PushConstants.PAYLOAD_FIELD] || '';
     try {
       payload = JSON.parse(payload);
     } catch(e) { }
-    if (payload) {
+    if (typeof payload === 'object') {
```
Comment (hermanliang) [inline src/dashboard/Push/PushIndex.react.js]: check type of payload

**Pull request merged**